aiida/engine/processes/process.py

```diff
@@ -745,6 +747,16 @@ def _setup_metadata(self, metadata: dict) -> None:
             else:
                 raise RuntimeError(f'unsupported metadata key: {name}')
 
+        # Store JSON-serializable values of ``netadata`` ports in the node's attributes. Note that instead of passing in
+        # the ``metadata`` inputs directly, the entire namespace of raw inputs is passed. The reason is that although
+        # currently in ``aiida-core`` all input ports that set ``metadata=True`` in the port specification are located
+        # within the ``metadata`` port namespace, this may not always be the case. The ``_filter_serializable_metadata``
```

Comment:
you might want to mention how you plan to use this rather than just that it _could_ be used.
(or just remember to update this comment when you add your next PR)

Reply:
This essentially pertains to my last comment regarding the last remaining open question. Should we allow other plugins to add `metadata` ports and have them store data in the process node's attributes? If so, then this is necessary. If we don't, then we may still want to keep this although it can be done differently. I am not sure what the answer to the question is though